enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,439 @@
+network customizations applied before the host can reach the Machine Config
+Server.
+
+### RHCOS live ISO Environment
+
+As of OCP 4.5, [RHCOS includes a new live ISO/PXE ramdisk
+environment](https://github.com/openshift/enhancements/pull/210) for installing
+RHCOS.  This environment is used to run
+[coreos-installer](https://github.com/coreos/coreos-installer/). Following that
+work, [RHCOS will include the ability to run a text-based UI to configure
+networking](https://github.com/openshift/enhancements/pull/291) in the ramdisk:
+`nmtui`.  This installation process also provides the ability to provide
+networking configuration via Ignition if the administrator does not want to
+follow an interactive configuration process.
+
+The `coreos-installer` also includes a
+[`--copy-network`](https://github.com/coreos/coreos-installer/pull/212) option
+which can be used to persist whatever network configuration was done using
+`nmtui` to the resulting installed host.  There is a [dracut
+module](https://github.com/coreos/fedora-coreos-config/pull/346) which will
+apply these settings in the initramfs.
+
+These enhancements provide some excellent improvements to early network
+configuration capabilities.  However, it is currently most beneficial to the
+UPI bare metal flow.  The IPI bare metal flow will need to do more work to
+figure out how to make use of these tools.  Bare Metal IPI makes use of a
+different deployment ramdisk that is part of the Ironic project.  While
+conceptually similar, it has some significant differences in implementation.
+Future integration work may provide better alignment between the Bare Metal IPI
+tooling and the RHCOS installer.
+
+### SR-IOV Operator
+
+Some clusters may want to allocate SR-IOV devices to workloads for a secondary
+network interface with high performance.  These devices are managed by the
+[SR-IOV Operator](https://github.com/openshift/sriov-network-operator).
+
+
+## Future Improvements
+
+### Disabling initramfs Networking
+
+Configuration of initramfs networking is lacking in some desired flexibility,
+as shown by the discussion in [this ignition
+issue](https://github.com/coreos/ignition/issues/979).  The default behavior in
+RHCOS is to attempt to get a DHCPv4 or DHCPv6 lease on one of the host’s
+interfaces.  This can cause a lot of undesirable delay if there are a lot of
+network interfaces present and DHCP is not available.  It will also cause
+booting to fail if DHCP never succeeds.
+
+One improvement is the ability to disable initramfs networking completely,
+which would be useful in combined with providing the host a full ignition
+configuration without needing to retrieve the bulk of it from a network
+location.  See “Flattened Ignition Configuration” for some more info on that
+piece.
+
+### NetworkManager in initramfs
+
+Fedora CoreOS has already moved to running NetworkManager in the initramfs.
+More info can be found in
+<https://github.com/coreos/fedora-coreos-tracker/issues/394>.
```

Comment:
It's worth noting that we already changed 4.6 to running NM in the inintramfs as well.

Reply:
Thanks, I wasn't sure if this had made it to RHCOS yet.

Reply:
Can you help me understand this and summarize it?  Is all of the dracut content still relevant, and it's just that NetworkManager is there and running in case additional network configuration is desired?

Reply:
Basically the user interface stays the same (network kargs defined in `man dracut.cmdline`) but the implementation is the `35network-manager` dracut module vs the `35network-legacy` dracut module, which uses the legacy network-scripts. It has some implications on the behavior though. For example [we get smarter timeout behavior](https://bugzilla.redhat.com/show_bug.cgi?id=1836248#c40) among others mentioned in https://github.com/coreos/fedora-coreos-tracker/issues/394#issuecomment-604598128.

Reply:
Thanks.  I've added notes on this in a new commit, mostly copying this helpful text.  :)